python/mlx/nn/layers/activations.py

```diff
@@ -162,6 +163,60 @@ def gelu_fast_approx(x):
     return x * mx.sigmoid(1.773 * x)
 
 
+def glu(x: mx.array) -> mx.array:
+    """Applies the gated linear unit function.
+
+    This function splits the last dimension of the input into two halves
+    (a and b) and applies :math:`a * \sigma(b)`.
+
+    .. math::
+        \\textrm{GLU}(x) = a * \sigma(b)
+    """
+    a, b = mx.split(x, indices_or_sections=2, axis=-1)
+    return a * mx.sigmoid(b)
+
+
+def gated_activation(x: mx.array, activation_function: Callable) -> mx.array:
+    """Applies gated activation function.
+
+    This function splits the last dimension of the input into two halves
+    (a and b) and applies :math:`f(a) * \sigma(b)`
+    where :math:`f` is the `activation_function` parameter.
+    """
+    a, b = mx.split(x, indices_or_sections=2, axis=-1)
+    return activation_function(a) * mx.sigmoid(b)
+
+
+@_make_activation_module(glu)
+class GLU(Module):
```

Comment:
If we give `glu` an `axis` param, we can add it as member variable here.

Reply:
Yes, I thought about adding `axis` param but at the same time I thought that the last dimension will always be the dimension that will be used for splitting so I found no need to add `axis` param, but I will add it if you think it will be needed.